src/main/java/winstone/Logger.java

```diff
@@ -6,160 +6,61 @@
+ * creates a new {@link PrintStream} and sets standard output (stdout) and standard error (stderr)
+ * to that new stream with {@link System#setOut} and {@link System#setErr}. If this takes place
+ * after JUL has initialized the root logger, the root logger will not be aware of the switch
+ * because its {@link ConsoleHandler} will have already read the old value of {@link System#err}.
+ * For this reason, parsing the {@code --logfile} option and redirecting the standard output
+ * (stdout) and standard error (stderr) streams must take place prior to JUL initialization.
  *
  * @author <a href="mailto:rick_knowles@hotmail.com">Rick Knowles</a>
  * @version $Id: Logger.java,v 1.8 2006/11/09 06:01:43 rickknowles Exp $
  */
 public class Logger {
 
-    public final static String DEFAULT_STREAM = "Winstone";
-    public final static Level MIN = Level.OFF;
-    public final static Level ERROR = Level.SEVERE;
-    public final static Level WARNING = Level.WARNING;
-    public final static Level INFO = Level.INFO;
-    public final static Level SPEED = Level.FINE;
-    public final static Level DEBUG = Level.FINER;
-    public final static Level FULL_DEBUG = Level.FINEST;
-    public final static Level MAX = Level.ALL;
-
-    protected final static Object semaphore = new Object();
+    private final static Object semaphore = new Object();
     static boolean initialised = false;
-    static Writer defaultStream;
-    static Map<String,Writer> namedStreams;
-//    protected static Collection nullStreams;
     static boolean showThrowingThread;
 
     /**
      * Initialises default streams
      */
     public static void init(Level level) {
-        init(level, System.out, Charset.defaultCharset(), false);
+        init(level, false);
     }
 
     public static void init(int level) {
         init(Level.parse(String.valueOf(level)));
     }
 
-    /**
-     * Initialises default streams
-     *
-     * @deprecated use {@link #init(Level, OutputStream, Charset, boolean)}
-     */
-    @Deprecated
-    public static void init(
-            Level level, OutputStream defaultStream, boolean showThrowingThreadArg) {
-        init(level, defaultStream, Charset.defaultCharset(), showThrowingThreadArg);
-    }
-
     /**
      * Initialize default streams
      */
     public static void init(
             Level level,
-            OutputStream defaultStream,
-            Charset defaultStreamCharset,
             boolean showThrowingThreadArg) {
         synchronized (semaphore) {
             if (!initialised) { // recheck in case we were blocking on another init
-                initialised = false;
-                LOGGER.setLevel(level);
-                namedStreams = new HashMap<>();
-//                nullStreams = new ArrayList();
-                initialised = true;
-                setStream(DEFAULT_STREAM, defaultStream, defaultStreamCharset);
+                java.util.logging.Logger.getLogger("").setLevel(level);
```

Comment:
Key change in semantics: setting the level on the root logger rather than just Winstone's logger. This enables the `--debug` argument to be mildly useful for Jenkins users.